app/code/Magento/Customer/Model/AccountManagement.php

```diff
@@ -928,7 +928,9 @@ public function createAccountWithPasswordHash(CustomerInterface $customer, $hash
             }
             $this->customerRegistry->remove($customer->getId());
         } catch (InputException $e) {
+            $this->registry->register('isSecureArea', true, true);
             $this->customerRepository->delete($customer);
+            $this->registry->unregister('isSecureArea');
```

Comment:
I believe this is not something we are looking to accept, simulation of the secure area here could bring security problems for the platform. Other suggestions to fix this issue would be the usage of transaction and rollback in case some exception has happened?

If the approach above is not suitable, we may need to rethink if the customer deletion here is actually something we are looking for when an exception happens from Address point of view.

Reply:
Yeah, if we can modify the code to make it possible to rollback customer creation, that would be the proper fix. (We don't want to create a customer if incorrect address data was given)

But I'm not sure if it's possible to rewrite this code to use DB connection directly. Take a look at the [customerRepository::save](https://github.com/magento/magento2/blob/HEAD/app/code/Magento/Customer/Model/ResourceModel/CustomerRepository.php#L200) that is used a few lines above to create a customer.

Reply:
You are right, such changes would have to be done on ResourceModel layer instead, it definitely would be more complicated. Another approach would be adapting the user interface instead, to be honest, logically the Customer does not depend on the address to exist and I believe it shouldn't be deleted if there was a problem on saving it's address, but this approach may also be complicated.

Reply:
Changing the logic to allow to successfully create an account may be the simplest solution (do not throw an exception from “catch” statement. The only thing that is not possible in this case is to inform the customer about specific error during account creation (invalid address field value).

I will test it tomorrow and let you know about the results.

Reply:
Works nicely. Here is a step-by-step result:

1. Press "register account" when trying to checkout with multishipping addresses.
2. Fill the form and press Register.
3. Account is successfully created and a user is redirected to multishipping's "Create Shipping Address" step.
4. Save the new address and proceed to the next checkout step.

The only downside is that the customer filled the address on the register account page, but it wasn't saved and no errors were given but I think it's not citical.

Reply:
Hi @vovayatsyuk, good stuff! Can you push the changes? I guess it will be easier for me to try locally!

Thanks!